src/ray/gcs/redis_async_context.h

```diff
@@ -92,8 +104,34 @@ class RedisAsyncContext {
   /// should be minimum.
   std::mutex mutex_;
   std::unique_ptr<redisAsyncContext, RedisContextDeleter> redis_async_context_;
-};
 
+  instrumented_io_context &io_service_;
+  boost::asio::ip::tcp::socket socket_;
+  // Hiredis wanted to add a read operation to the event loop
+  // but the read might not have happened yet
+  bool read_requested_{false};
+  // Hiredis wanted to add a write operation to the event loop
+  // but the read might not have happened yet
+  bool write_requested_{false};
+  // A read is currently in progress
+  bool read_in_progress_{false};
+  // A write is currently in progress
+  bool write_in_progress_{false};
+
+  void operate();
```

Comment:
Any chance for documentation? `operate`, `cleanup`, and most of the function names are confusing without inspecting the code, please add some documentations, thanks.

Reply:
Acutally I just copy them from the original `asio.h`. There is also no comment there. But I'll try to add some comments.

Reply:
Thank you!